src/Form/Control/ScopeBuilder.php

```diff
@@ -333,17 +335,43 @@ protected function buildQuery(Model $model)
      */
     protected function addFieldRule(Field $field): self
     {
+
         $type = ($field->enum || $field->values || $field->reference) ? 'enum' : $field->type;
+        $typeOptions = $type ? $this->getTypeOptions($type) : [];
 
         $this->rules[] = self::getRule($type, array_merge([
             'id' => $field->short_name,
             'label' => $field->getCaption(),
             'options' => $this->options[strtolower((string) $type)] ?? [],
-        ], $field->ui['scopebuilder'] ?? []), $field);
+        ], $field->ui['scopebuilder'] ?? [], $typeOptions), $field);
 
         return $this;
     }
 
+    protected function getTypeOptions(string $type)
```

Comment:
@georgehristov - I refactored ScopeBuilder using this rule['options'] instead of settings props to date, datetime or time type. This way, we decide to use other Vue components in QueryBuilder, then we could simply pass component props using rule.options. 

I have also set dateFormat accoding to ui_persistence, which means that value sent to the server is format accordingly. Let me know if you need to change, i.e. if value sent always need to be in the same format.

Reply:
@georgehristov - Just refactor (again) to use component type instead of field type. and change $rule['options'] to $rule['componentProps'] I think it make more sense this way.

Reply:
Now use altFormat for date display and dedicated format for setting string value.